devito/core/gpu_openacc.py

```diff
@@ -0,0 +1,171 @@
+from functools import partial
+
+import cgen as c
+
+from devito.core.gpu_openmp import (DeviceOpenMPNoopOperator, DeviceOpenMPIteration,
+                                    DeviceOmpizer, DeviceOpenMPDataManager)
+from devito.exceptions import InvalidOperator
+from devito.logger import warning
+from devito.passes.iet import optimize_halospots, mpiize, hoist_prodders
+from devito.tools import as_tuple, timed_pass
+
+__all__ = ['DeviceOpenACCNoopOperator', 'DeviceOpenACCOperator',
+           'DeviceOpenACCCustomOperator']
+
+
+# TODO: currently inhereting from the OpenMP Operators. Ideally, we should/could
+# abstract things away so as to have a separate, language-agnostic superclass
+
+
+class DeviceOpenACCIteration(DeviceOpenMPIteration):
+
+    @classmethod
+    def _make_construct(cls, **kwargs):
+        return 'acc parallel loop'
+
+
+class DeviceAccizer(DeviceOmpizer):
+
+    lang = {
+        'atomic': c.Pragma('acc atomic update'),
+        'map-enter-to': lambda i, j:
+            c.Pragma('acc enter data copyin(%s%s)' % (i, j)),
+        # 'map-enter-alloc': lambda i, j:
+        #    c.Pragma('omp target enter data map(alloc: %s%s)' % (i, j)),
```

Comment:
Leftover debug?

Reply:
As far as I know there is no explicit correspondence bewtween openacc and openmp in this directive

Reply:
yes, more of a memo. Anyway, already dropped in the branch on top of this